app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,28 @@
+class OmniauthCallbacksController < Devise::OmniauthCallbacksController
+
+  def do_omniauth
+    @user = User.from_omniauth(request.env["omniauth.auth"])
+    provider_data = @user.provider
+    provider_kind = @user.provider.capitalize
+
+    if @user.provider == "google_oauth2"
+      provider_kind = "Google"
+    end
+
+    if @user.persisted?
+      flash[:notice] = I18n.t("devise.omniauth_callbacks.success", kind: "#{provider_kind}")
+      sign_in_and_redirect @user, :event => :authentication
+    else
+      session["devise.#{provider_data}_data"] = request.env["omniauth.auth"]
+      redirect_to new_user_registration_url
+    end
+  end
+
+  alias_method :facebook, :do_omniauth
+  alias_method :google_oauth2, :do_omniauth
+
+  def failure
+    flash[:alert] = "Could not authenticate."
```

Comment:
Could you make this message a bit more user-friendly? I'm thinking something along the lines of "Authentication failed. Please try again later or sign up with MANIAC Chess directly by providing an email and password."